src/Identity/UI/src/Areas/Identity/Pages/V3/Account/Manage/Email.cshtml.cs

```diff
@@ -120,7 +120,7 @@ public override async Task<IActionResult> OnPostAsync()
         {
             if (!ModelState.IsValid)
             {
-                return Page();
+                return await OnGetAsync();
```

Comment:
Same comment as below, I don't understand all these changes to OnGetAsync()

Reply:
My suggestion would be to factor out lines 100-114 into a private method. 

You basically want to do all of the initialization that's required to show the view here, and that's accomplished in a slightly non-standard way by calling `OnGetAsync` directly. I think making a private method would be more clear.